db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryArithmeticOperatorNode.java

```diff
@@ -363,4 +359,56 @@ public double getBaseOperationCost() throws StandardException {
         }
         return localCost + super.getBaseOperationCost() + SIMPLE_OP_COST * FN_CALL_COST_FACTOR;
     }
+
+    /** @see ValueNode#evaluateConstantExpressions */
+    @Override
+    ValueNode evaluateConstantExpressions() throws StandardException {
+        if (getLeftOperand() instanceof ConstantNode && getRightOperand() instanceof ConstantNode) {
+            ConstantNode lhs = (ConstantNode) getLeftOperand();
+            ConstantNode rhs = (ConstantNode) getRightOperand();
+
+            if (lhs.isNull()) {
+                return lhs;
+            }
+            if (rhs.isNull()) {
+                return rhs;
+            }
+            if (lhs instanceof NumericConstantNode && rhs instanceof NumericConstantNode) {
+                DataTypeDescriptor resultType = lhs.getTypeCompiler().resolveArithmeticOperation(
+                        lhs.getTypeServices(),
+                        rhs.getTypeServices(),
+                        operator
+                );
+                NumberDataValue result;
+                NumberDataValue leftValue = (NumberDataValue) lhs.getValue();
+                NumberDataValue rightValue = (NumberDataValue) rhs.getValue();
+                NumberDataValue receiver = leftIsReceiver() ? leftValue : rightValue;
+                switch (operator) {
+                    case TypeCompiler.TIMES_OP:
+                        result = receiver.times(leftValue, rightValue, null);
+                        break;
```

Comment:
A general question is what's the behavior if left + right overflows. If in runtime there is a SQL warning, we wouldn't be able to set the warning at this step.

But I guess that's something we can discuss later.